### ISSUE TITLE
Hi Sprinter, thank for your code, i have some questions as follows.

### ISSUE THREAD

**Issue by LanXiaoPang613** [Sprinter1999/FedLSR#3]
1. The L_e in the code i think is the **Loss_reg = 𝐿𝑜𝑠𝑠𝑟𝑒𝑔 = 𝐽𝑆 (𝑞1, 𝑞2) =1/2*(𝐾𝐿(𝑞1 ∥𝑈 ) + 𝐾𝐿(𝑞2 ∥𝑈 )).** But when i read the code, i find it is calculated by**" L_e = - (torch.mean(torch.sum(sm(logits1) * lsm(logits1), dim=1)) + torch.mean(torch.sum(sm(logits1) * lsm(logits1), dim=1))) * 0.5 "**. It seems ignore the " -sum(q1*logU)+sum(q2*logU)"???

**Comment by Sprinter1999** [comment]
Thank you LanXiao, L_e is in short of Loss of entropy, which is different from Loss_reg, in this demo code, Loss_reg is implemented in a single function. You can take a closer step at the code : ）, actually L_e can make FedLSR more robust but may slow down the convergence, you may check the discussion part of this paper.
Thank you again~

**Comment by LanXiaoPang613** [comment]
Ahhh, i have found it, thx, actually the L_e is come from "Joint opt" and "divide-Mix"? A really interesting trial.

**Comment by Sprinter1999** [comment]
hhh, i think it's from RobustFL, btw, it's a usual trick in semi-supervised learning : )

**Comment by LanXiaoPang613** [comment]
hhh, are you either the postgraduate or undergraduate the of China University of Science and Technology？outstanding ~~~

**Comment by LanXiaoPang613** [comment]
hhh, are you either the postgraduate or undergraduate the of China University of Science and Technology？outstanding ~~~

**Comment by Sprinter1999** [comment]
Thank you LanXiao, I'm second year PhD student at Chinese Academy of Sciences. You're outstanding too, and nice to meet you.
